### ISSUE TITLE
Better error handling in daml-helper.

### ISSUE THREAD

**Issue by associahedron** [digital-asset/daml#1221]
Three improvements to make:

1. When the user passes bad arguments to a daml-helper command, it should say `daml` not `daml-helper` in the error output from optparse-applicative, for example:

```
$ daml-head new --not-a-real-option
Invalid option `--not-a-real-option'

Usage: daml-helper new (--list | TARGET_PATH [TEMPLATE])

Available options:
  --list                   List the available project templates.
  TARGET_PATH              Path where the new project should be located
  TEMPLATE                 Name of the template used to create the project
                           (default: skeleton)
  -h,--help                Show this help text
```

2. When an exception occurs, the output should be readable. For example, here's what is currently displayed when running `daml start` outside of a project. 

```
$ daml-head start
daml-helper: DamlHelperError {errMessage = "Must be called from within a project", errInternal = Nothing}
```

**Comment by hurryabit** [comment]
I'm not the biggest fan of the `TARGET_PATH [TEMPLATE]` thing. It's easy to mess up the order of the two strings. How about adding `--template TEMPLATE` instead?

**Comment by associahedron** [comment]
> I'm not the biggest fan of the `TARGET_PATH [TEMPLATE]` thing. It's easy to mess up the order of the two strings. How about adding `--template TEMPLATE` instead?

Not a fan, sorry.

If you mess up the order, you will almost always get an error, especially after my last PR #1212.

This is also very off topic for this issue.

**Comment by cocreature** [comment]
@associahedron You already changed sandbox to use `run-jar` https://github.com/digital-asset/daml/blob/9f184244f9e3951222c8d2376741845e62b2a34b/release/sdk-config.yaml.tmpl#L29

**Comment by associahedron** [comment]
> @associahedron You already changed sandbox to use `run-jar`
> 
> https://github.com/digital-asset/daml/blob/9f184244f9e3951222c8d2376741845e62b2a34b/release/sdk-config.yaml.tmpl#L29

:sweat_smile:  Brilliant. I'm going to remove that from the list and raise a separate issue w/ the sandbox team.

**Comment by associahedron** [comment]
Actually @gaboraranyossy-da is on it already #1217 :+1:

**Comment by cocreature** [comment]
Fixed in #1256 and #1258
